Sources/ParseSwift/Coding/ParseEncoder.swift

```diff
@@ -0,0 +1,259 @@
+    }
+
+    mutating func encode<T>(_ value: T, forKey key: Key) throws where T: Encodable {
+        if skippedKeys.contains(key.stringValue) { return }
+
+        dictionary[key.stringValue] = try _ParseEncoder.encode(
+            value,
+            with: codingPath + [key],
+            skippingKeys: skippedKeys
+        )
+    }
+
+    mutating func nestedContainer<NestedKey>(
+        keyedBy keyType: NestedKey.Type,
+        forKey key: Key
+    ) -> KeyedEncodingContainer<NestedKey> where NestedKey: CodingKey {
+        let container = _ParseEncoderKeyedEncodingContainer<NestedKey>(
+            codingPath: codingPath + [key],
+            dictionary: dictionary,
+            skippingKeys: skippedKeys
+        )
+
+        return KeyedEncodingContainer(container)
+    }
+
+    mutating func nestedUnkeyedContainer(forKey key: Key) -> UnkeyedEncodingContainer {
+        _ParseEncoderUnkeyedEncodingContainer(
+            codingPath: codingPath + [key],
+            dictionary: dictionary,
+            skippingKeys: skippedKeys
+        )
+    }
+
+    mutating func superEncoder() -> Encoder {
+        fatalError()
+    }
+
+    mutating func superEncoder(forKey key: Key) -> Encoder {
+        fatalError()
+    }
+}
+
+// MARK: _ParseEncoderSingleValueEncodingContainer
+internal struct _ParseEncoderSingleValueEncodingContainer: SingleValueEncodingContainer {
+    let codingPath: [CodingKey]
+    let dictionary: NSMutableDictionary
+    let skippedKeys: Set<String>
+
+    init(codingPath: [CodingKey], dictionary: NSMutableDictionary, skippingKeys: Set<String>) {
+        self.codingPath = codingPath
+        self.dictionary = dictionary
+        self.skippedKeys = skippingKeys
+    }
+
+    var key: String {
+        codingPath.last?.stringValue ?? "<root>"
+    }
+
+    mutating func encodeNil() throws {
+        dictionary[key] = nil
+    }
+
+    mutating func encode<T>(_ value: T) throws where T: Encodable {
+        dictionary[key] = try _ParseEncoder.encode(value, with: codingPath, skippingKeys: skippedKeys)
+    }
+}
+
+// MARK: _ParseEncoderUnkeyedEncodingContainer
+internal struct _ParseEncoderUnkeyedEncodingContainer: UnkeyedEncodingContainer {
+    let codingPath: [CodingKey]
+    let dictionary: NSMutableDictionary
+    let skippedKeys: Set<String>
+
+    var array: NSMutableArray {
+        get {
+            // swiftlint:disable:next force_cast
+            dictionary[key] as! NSMutableArray
+        }
+
+        set { dictionary[key] = newValue }
+    }
+
+    var count: Int {
+        array.count
+    }
+
+    var key: String {
+        codingPath.last?.stringValue ?? "<root>"
+    }
+
+    init(codingPath: [CodingKey], dictionary: NSMutableDictionary, skippingKeys: Set<String>) {
+        self.codingPath = codingPath
+        self.dictionary = dictionary
+        self.skippedKeys = skippingKeys
+
+        self.array = NSMutableArray()
+    }
+
+    mutating func encodeNil() throws {
+        array.add(NSNull())
```

Comment:
NSNull? I haven't really dug into the Foundation encoder, but is this typical? I would have thought they would have moved away from these types of things. I can see it being used in Unarchiver/Archiver

Reply:
You can't add `nil` to an `NSArray`, so you have to add `NSNull` instead.